src/boot/efi/boot.c

```diff
@@ -2354,28 +2354,38 @@ static EFI_STATUS secure_boot_enroll(EFI_FILE *root_dir, Config *config, ConfigE
         clear_screen(COLOR_NORMAL);
         ps_string(L"Enrolling secure boot keys from directory: %s\n", entry->current_name);
 
-        Print(L"*** DANGER ***\n");
-        Print(L"Enrolling Secure Boot keys might soft-brick your machine.\n");
-        Print(L"However the option i-might-brick-my-machine-please-auto-enroll-keys has been set\n");
-        Print(L"The enrollement of secure boot keys will proceed after %d seconds\n", config->timeout_sec);
-        Print(L"*** DANGER ***\n");
-
-        Print(L"\nType q, Q or ESC to cancel.\n\n");
+        CHAR8 yes[1] = { 1 };
+        err = efivar_set_raw(EFI_GLOBAL_GUID, L"AuditMode", &yes, sizeof(yes), EFI_VARIABLE_NON_VOLATILE |
+                EFI_VARIABLE_BOOTSERVICE_ACCESS |
+                EFI_VARIABLE_RUNTIME_ACCESS);
+        if (EFI_ERROR(err)) {
+                if (!config->allow_unsafe_enrollement)
+                        return log_error_status_stall(err, L"Failed switching to secure boot audit mode: %r", err);
+                else {
```

Comment:
given the `if` branch returned you can drop the `else {` here, and save one level of indentation